### PR TITLE
Avoid duplicate inference results for `Tuple[Optional[int], ...]`

### PR DESCRIPTION
**Pull request by jacobtylerwalls**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description
Follow-up to #2185. Should help reduce noise in the pylint primer.

### CODE REVIEW

**Comment by jacobtylerwalls** [comment]
Blocked by a pylint patch release

**Review by DanielNoord — APPROVED** [review]

**Comment by jacobtylerwalls** [comment]
@Pierre-Sassoulas I'd like to unblock this PR so we can get an astroid patch release out and start getting better pylint primer results. We need a pylint patch release with the isort changes to do it, or we could just temporarily pin pylint to main in astroid CI. I'm leaning toward the former, since it's been a couple months since a pylint patch release. WDYT?

This would mean doing things in the reverse order we generally do them (first the pylint patch, then the astroid patch), but then we can upgrade astroid in pylint main without releasing.

**Comment by Pierre-Sassoulas** [comment]
I was trying to prepare a pylint patch release "soon" ™️, let's release then. (What is in the current milestone are not blockers, only nice to have.)

**Pull request merged**
